custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -0,0 +1,235 @@
+        episode = get_episode_case_from_voucher(voucher_case.domain, voucher_case.case_id)
+        episode_case_properties = episode.dynamic_case_properties()
+
+        def _get_voucher_count():
+            # This is an expensive operation, so only call this function if all other conditions are true.
+            voucher_count = episode_case_properties.get('approved_voucher_count', 0)
+            if voucher_count < 2:
+                voucher_count = len(
+                    get_approved_prescription_vouchers_from_episode(episode.domain, episode.case_id)
+                )
+            return voucher_count
+
+        not_sent = voucher_case_properties.get("event_{}".format(DRUG_REFILL_EVENT)) != "sent"
+
+        return (
+            # TODO: Confirm state == "fulfilled"
+            voucher_case_properties.get("state") == "fulfilled"
+            and voucher_case_properties.get("type") == "prescription"
+            and not_sent
+            and case_properties_changed(voucher_case, ['state'])
+            and is_valid_voucher_submission(voucher_case)
+            and _get_voucher_count() >= 2
+        )
+
+
+class BETSSuccessfulTreatmentRepeater(BaseBETSRepeater):
+    friendly_name = _("BETS - Patients: Cash transfer on successful treatment completion (episode case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETSSuccessfulTreatmentRepeaterView
+        return reverse(BETSSuccessfulTreatmentRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        if not self.case_types_and_users_allowed(episode_case):
+            return False
+
+        case_properties = episode_case.dynamic_case_properties()
+        not_sent = case_properties.get("event_{}".format(SUCCESSFUL_TREATMENT_EVENT)) != "sent"
+        return (
+            case_properties.get("treatment_outcome") in ("cured", "treatment_completed")
+            and case_properties_changed(episode_case, ["treatment_outcome"])
+            and not_sent
+            and is_valid_archived_submission(episode_case)
+        )
+
+
+class BETSDiagnosisAndNotificationRepeater(BaseBETSRepeater):
+    friendly_name = _("BETS - Providers: For diagnosis and notification of TB case (episode case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETSDiagnosisAndNotificationRepeaterView
+        return reverse(BETSDiagnosisAndNotificationRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        if not self.case_types_and_users_allowed(episode_case):
+            return False
+
+        case_properties = episode_case.dynamic_case_properties()
+        not_sent = case_properties.get("event_{}".format(DIAGNOSIS_AND_NOTIFICATION_EVENT)) != "sent"
+        return (
+            case_properties.get("pending_registration") == "no"
```

Comment:
this case property is actually called `private_sector_episode_pending_registration`. Sorry about that. I've updated the spec too.